packages/snap/src/features/send/views/SendForm/events.tsx

```diff
@@ -64,15 +64,28 @@ async function onSourceAccountSelectorValueChange({
   event: InputChangeEvent;
   context: SendContext;
 }) {
-  context.fromAccountId = event.value as string;
+  const accountData = event.value as { accountId: string; addresses: string[] };
+  context.fromAccountId = accountData.accountId;
   context.error = null;
+
+  context.validation = {};
+
   context.validation[SendFormNames.SourceAccountSelector] =
     validateField<SendFormNames>(
       SendFormNames.SourceAccountSelector,
       context.fromAccountId,
       validation(context),
     );
 
+  if (context.toAddress) {
+    context.validation[SendFormNames.DestinationAccountInput] =
+      validateField<SendFormNames>(
+        SendFormNames.DestinationAccountInput,
+        context.toAddress,
+        validation(context),
+      );
+  }
```

Comment:
Do we need to validate address here?

Reply:
Same as above, but is now fixed and not needed.